storage/s3.go

```diff
@@ -43,19 +42,9 @@ const (
 )
 
 // Re-used AWS sessions dramatically improve performance.
-var cachedSession func() *session.Session
-
-// Init creates a new global S3 session.
-func Init(opts Options) error {
-	sess, err := newSession(opts)
-	if err != nil {
-		return err
-	}
-
-	cachedSession = func() *session.Session {
-		return sess
-	}
-	return nil
+var sessionProvider = &s3Session{
+	Mutex:    sync.Mutex{},
```

Comment:
Zero value of sync.Mutex is an unlocked mutex, thus creating a new value seems unnecessary.